### PR TITLE
layers: Fix to the logic defining a `VkStridedDeviceAddressRegionKHR` as unused

### PR DESCRIPTION
**Pull request by arno-lunarg**
- Closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/5635

I got my info from here: https://gitlab.khronos.org/vulkan/vulkan/-/issues/3464#note_407581

### CODE REVIEW

**Review by artem-lunarg — APPROVED** [review]
LGTM! Thanks for clarifying this.

File: tests/positive/ray_tracing.cpp
Comment (juan-lunarg) [inline tests/positive/ray_tracing.cpp:1]: Revert change to this file. Or add change in 2nd commit.

File: tests/negative/ray_tracing.cpp [hunk 1]
```diff
@@ -1089,10 +1089,20 @@ TEST_F(VkLayerTest, RayTracingValidateCmdTraceRaysKHR) {
...
+    }
+
+    // pRayGenShaderBindingTable->size == 0, deviceAddress is invalid => region is considered unused so no error
+    {
+        VkStridedDeviceAddressRegionKHR empty_region = stridebufregion;
+        empty_region.deviceAddress += buffer.create_info().size + 128;
+        empty_region.size = 0;
+        empty_region.stride = 0;
+        vkCmdTraceRaysKHR(m_commandBuffer->handle(), &empty_region, &stridebufregion, &stridebufregion, &stridebufregion, 100, 100,
+                          1);
```
Comment (juan-lunarg) [inline tests/negative/ray_tracing.cpp:1105]: Move positive testing to `tests/positive/ray_tracing.cpp`.

**Review by juan-lunarg — CHANGES_REQUESTED** [review]
LGTM overall

**Review by juan-lunarg — APPROVED** [review]

**Pull request merged**